test/tsclient.c

```diff
@@ -310,6 +468,68 @@ pgmoneta_tsclient_read(block_ref_table** brt)
    return 1;
 }
 
+int
+pgmoneta_tsclient_summarize_wal(char* server)
+{
+   char* wal_dir = NULL;
+   char* summary_dir = NULL;
+   struct main_configuration* config = NULL;
+   int srv = -1;
+   xlog_rec_ptr s_lsn;
+   xlog_rec_ptr e_lsn;
+
+   config = (struct main_configuration*)shmem;
+   /* get the server index */
+   for (int i = 0; i < config->common.number_of_servers; i++)
+   {
+      if (!strcmp(config->common.servers[i].name, server))
+      {
+         srv = i;
+         break;
+      }
+   }
+
+   /* get the starting lsn for summary */
+   if (do_checkpoint_and_get_lsn(srv, &s_lsn))
+   {
+      goto error;
+   }
+
+   /* generate wal records using pgbench */
+   if (pgmoneta_tsclient_execute_pgbench(srv, false, 0, 0, 0))
```

Comment:
It's better to just insert a record into a table in my opinion. You'll have a clearer understanding wether the summarizer is working as expected -- by seeing if it picks up the changes on the table relation. pg_bench can be used if want to stress test the summarizer to see its performance.